addons/survey/tests/test_survey_performance.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import time
+
+from logging import getLogger
+
+from odoo.addons.survey.tests import common
+from odoo.addons.website.tests.test_performance import UtilPerf
+from odoo.tests import tagged
+from odoo.tests.common import users, warmup
+
+_logger = getLogger(__name__)
+
+
+@tagged('post_install', '-at_install')
+class SurveyPerformance(common.TestSurveyCommon, UtilPerf):
+
+    @warmup
+    @users('survey_manager')
+    def test_survey_results_with_filters(self):
+        url = '/survey/results/%i?filters=multiple_choice,0,6|numerical_box,0,16' % self.ref("survey.survey_feedback")
+
+        t0 = time.time()
+        with self.assertQueryCount(default=48):
+            self.url_open(url)
+        t1 = time.time()
+        _logger.info('Browsed %s, time %.3f', url, t1 - t0)
```

Comment:
I would also add one or two more complex cases here too (without the timing part each time) to see how efficient we made the search for multiple filters on the same model.

Reply:
Hello :slightly_smiling_face: 

If I do this :
```python
    @warmup
    @users('survey_manager')
    def test_survey_results_with_filters(self):
        url = '/survey/results/%i?filters=A,0,57|L,0,107' % self.ref("survey.survey_demo_quiz")

        t0 = time.time()
        with self.assertQueryCount(default=48):
            self.url_open(url)
        t1 = time.time()
        _logger.info('Browsed %s, time %.3f', url, t1 - t0)

        url = '/survey/results/%i?filters=A,0,57' % self.ref("survey.survey_demo_quiz")

        t0 = time.time()
        with self.assertQueryCount(default=48):
            self.url_open(url)
        t1 = time.time()
        _logger.info('Browsed %s, time %.3f', url, t1 - t0)

        url = '/survey/results/%i?filters=A,0,57|A,0,54' % self.ref("survey.survey_demo_quiz")

        t0 = time.time()
        with self.assertQueryCount(default=48):
            self.url_open(url)
        t1 = time.time()
        _logger.info('Browsed %s, time %.3f', url, t1 - t0)
```
I get this :
![image](https://user-images.githubusercontent.com/107679666/215459595-26119767-2ca2-467e-b15c-ba97fa7f9eab.png)

We can see that the `A,0,57` filter takes 0,173 after the warmup and the filters `A,0,57|A,0,54` take 0,179 so roughly the same amount of time :slightly_smiling_face: That's great ! But the query count stays at 48 for every one of the test case. :thinking: Is it what was expected ? If I understand correctly this should have taken 96 queries with the previous algorithm? Shouldn't we keep the timing part for each test case then? as it is what's actually impacted here :thinking: 

Waiting for your opinion :slightly_smiling_face: Thanks!!

Reply:
I'll answer quickly for the different parts, we can discuss further in Discord!
1. We could keep one more to be able to see that adding filters does not impact the time to make the answer but it is not useful to have it for all cases, and there is too much possible variability that we likely won't be able to `assert` anything on them anyway.
3. If you want to compare with the previous implementation, you could keep the same tests in a different branch with the other commit and run these tests about query counts (If less efficient, the tests should fail). However, the 48 queries are not only those from our `_extract_filters_data` method, so the difference will not be as large. 
4. I would also here define `id` attributes in `setup, and use them in the test cases. E.g., 

```suggestion
    def setUp(self)
        self.survey_feedback_id = self.ref("survey.survey_feedback").id
        self.q2 = self.ref("survey.question_2")
        self.q2_a2_id = self.ref("survey.question_2_suggested_answer_2").id
        self.q2_a2_filter = f"numerical_box,0,{self.q2_a2_id}"
        ...

    def test_check_demo_data():
        self.assertEqual(self.q2.question_type, "numerical_box")
        ...

    @warmup
    @users('survey_manager')
    def test_survey_results_with_filters(self):
        # no magic numbers or additional queries here
        url = f'/survey/results/{self.survey_feedback_id}?filters={q2_a2_filter}'

        t0 = time.time()
        with self.assertQueryCount(default=48):
            self.url_open(url)
        t1 = time.time()
        _logger.info('Browsed %s, time %.3f', url, t1 - t0)
```